libs/orders/src/lib/order-hooks/use-order-edit.tsx

```diff
@@ -0,0 +1,126 @@
+import { useApolloClient } from '@apollo/client';
+import { determineId, removeDecimal } from '@vegaprotocol/react-helpers';
+import type {
+  OrderAmendmentBody,
+  OrderAmendmentBodyOrderAmendment,
+} from '@vegaprotocol/vegawallet-service-api-client';
+import { useState, useCallback, useEffect, useRef } from 'react';
+import type { VegaWalletOrderTimeInForce } from '@vegaprotocol/wallet';
+import { useVegaTransaction, useVegaWallet } from '@vegaprotocol/wallet';
+import { ORDER_EVENT_SUB } from './order-event-query';
+import type { Subscription } from 'zen-observable-ts';
+import type {
+  OrderEvent_busEvents_event_Order,
+  OrderEvent,
+  OrderEventVariables,
+} from './__generated__';
+import * as Sentry from '@sentry/react';
+
+export const useOrderEdit = () => {
+  const { keypair } = useVegaWallet();
+  const { send, transaction, reset: resetTransaction } = useVegaTransaction();
+  const [updatedOrder, setUpdatedOrder] =
+    useState<OrderEvent_busEvents_event_Order | null>(null);
+  const client = useApolloClient();
+  const subRef = useRef<Subscription | null>(null);
+
+  const reset = useCallback(() => {
+    resetTransaction();
+    setUpdatedOrder(null);
+    subRef.current?.unsubscribe();
+  }, [resetTransaction]);
+
+  useEffect(() => {
+    return () => {
+      resetTransaction();
+      setUpdatedOrder(null);
+      subRef.current?.unsubscribe();
+    };
+  }, [resetTransaction]);
+
+  const edit = useCallback(
+    async (order) => {
+      if (!keypair) {
+        return;
+      }
+
+      setUpdatedOrder(null);
+
+      try {
+        const res = await send({
+          pubKey: keypair.pub,
+          propagate: true,
+          orderAmendment: {
+            orderId: order.id,
+            marketId: order?.market?.id,
+            price: {
+              value: removeDecimal(
+                order.price,
+                order.market.decimalPlaces ?? 0
+              ),
+            },
+            timeInForce:
+              `TIME_IN_FORCE_${order.timeInForce}` as VegaWalletOrderTimeInForce,
```

Comment:
suggestion (non blocking): You might be able to use `VegaWalletTimeInForce[order.timeInForce]` which I think its a bit cleaner